### DISCUSSION TITLE
Build keeps failing

### DISCUSSION THREAD

**Discussion by Adarack** [PlummersSoftwareLLC/NightDriverStrip#49] [Q&A]
I can't seem to get anything other than the Demo to build and upload.  I'm attempting to get the Atom Lamp config to build for a Wemos D1 Mini ESP32. 

The platformio.ini I'm using is.

`[platformio]
data_dir    = ./data
default_envs = demowemos

[common]
lib_deps        = crankyoldgit/IRremoteESP8266  @ ^2.7.20
                  fastled/FastLED               @ ^3.4.0
                  adafruit/Adafruit BusIO       @ ^1.9.1
                  adafruit/Adafruit GFX Library @ ^1.10.12
                  bblanchon/ArduinoJson         @ ^6.8.14
                  olikraus/U8g2                 @ ^2.28.8
                  joaolopesf/RemoteDebug        @ ^3.0.5
                  kosme/arduinoFFT              @ ^1.5.6
                  me-no-dev/AsyncTCP            @ ^1.1.1
                  thomasfredericks/Bounce2      @ ^2.7.0
                  me-no-dev/ESP Async WebServer @ ^1.2.3
[env]
platform        = https://github.com/platformio/platform-espressif32.git
framework       = arduino
build_type      = debug
build_unflags   = -std=gnu++11

lib_deps        = ${common.lib_deps}

; This partition table attempts to fit everything in 4M of flash.
board_build.partitions = partitions_custom.csv

monitor_filters = esp32_exception_decoder
[env:demowemos]
platform      = espressif32
board         = esp32dev
monitor_speed = 115200
upload_speed  = 921600
upload_port   = COM11                    ; This will vary by machine
build_flags   = -DATOMLIGHT=1
                -std=gnu++17
                -Ofast `

And the faults I'm getting are below
 
`In file included from include/effects/particles.h:44:0,
                 from include/effects/stareffect.h:46,
                 from src/effects.cpp:36:
include/effects/faneffects.h: In function 'int GetFanPixelOrder(int, PixelOrder)':
include/effects/faneffects.h:117:26: error: 'FanPixelsVertical' was not declared in this scope
       return fanBase + ((FanPixelsVertical[fPos] + LED_FAN_OFFSET_BU) % FAN_SIZE);
                          ^
src/effects.cpp: At global scope:
src/effects.cpp:379:9: error: expected type-specifier before 'ChannelBeatEffect'
     new ChannelBeatEffect("ChannelBeat"),
         ^
src/effects.cpp:379:9: error: expected '}' before 'ChannelBeatEffect'
src/effects.cpp:379:9: error: expected ',' or ';' before 'ChannelBeatEffect'
src/effects.cpp:549:1: error: expected declaration before '}' token
 };
 ^
*** [.pio\build\demo2\src\effects.cpp.o] Error 1
========================================================== [FAILED] Took 20.03 seconds ==========================================================

Environment    Status    Duration
-------------  --------  ------------
demowemos          FAILED    00:00:20.028` 

Any help would be appreciated. 
Thank you

**Comment by upkeepdavid** [comment]
a D1 mini is an 8266

**Reply by Adarack** [reply]
This is the one I have. It's listed as ESP32. Also esptool output is  
`Detecting chip type... ESP32
Chip is ESP32-D0WDQ6 (revision 1)`
https://www.amazon.com/ACEIRMC-ESP-WROOM-32-Bluetooth-Development-Compatible/dp/B08L5XFWN6/ref=pd_sbs_1/130-6686194-1440163?pd_rd_w=PPPq8&pf_rd_p=0a3ad226-8a77-4898-9a99-63ffeb1aef90&pf_rd_r=5AAQC91EJD1MNWWTC5TZ&pd_rd_r=739d2f1c-f962-4b89-9f83-096f2f9f5210&pd_rd_wg=ZCXbX&pd_rd_i=B08L5XFWN6&psc=1

**Reply by BeldrothTheGold** [reply]
Regardless, the board configured in platformio.ini is esp32dev.

**Reply by Adarack** [reply]
> Regardless, the board configured in platformio.ini is esp32dev.

You are correct, I believe it should actually be

`[env:wemos_d1_mini32_atom]
platform = espressif32
board = wemos_d1_mini32
build_flags   = -DATOMLIGHT=1
                -std=gnu++17
                -Ofast `

**Comment by BeldrothTheGold** [comment]
Some of the other projects definitely have issues compiling. 
Most likely this is being caused by some expected define not being set.

**Reply by Adarack** [reply]
> Some of the other projects definitely have issues compiling. Most likely this is being caused by some expected define not being set.

It seems that even with audio disabled there are several places that keep looking for "gVURatio". Had to keep trying and commenting out the errors that called for it since I don't plan to use that feature at this time.

The FanPixelsVertical error was "fixed". Per this post I set the #define FAN_SIZE   16, not sure how this will act after being installed in the lamp though.
https://github.com/PlummersSoftwareLLC/NightDriverStrip/discussions/51#discussioncomment-1495317

So while not really fixed, I can at least build it now. I'm not enough of a coder to be able to actually correct these issues with the code though.
